src/test/kotlin/interactor/GetMostSuitableCityInteractorTest.kt

```diff
@@ -1,74 +1,64 @@
 package interactor
 
+import dataSource.CitiesHasNullFakeDataSource
+import dataSource.CitiesEmptyList
+import dataSource.MostSuitableCity
 import dataSource.FakeDataSource
-import model.CityEntity
 import org.junit.jupiter.api.Test
-
 import org.junit.jupiter.api.Assertions.*
-import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.TestInstance
 import org.junit.jupiter.api.function.Executable
 
+
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 internal class GetMostSuitableCityInteractorTest {
 
-    private lateinit var getMostSuitableCityInteractor: GetMostSuitableCityInteractor
     private lateinit var fakeDataSource: FakeDataSource
+    private lateinit var citiesEmptyList: CitiesEmptyList
+    private lateinit var interactor: GetMostSuitableCityInteractor
+    private lateinit var citiesHasNullFakeDataSource: CitiesHasNullFakeDataSource
 
-    @BeforeAll
+
+    @BeforeEach
     fun setUp() {
-        getMostSuitableCityInteractor = GetMostSuitableCityInteractor()
         fakeDataSource = FakeDataSource()
+        citiesEmptyList = CitiesEmptyList()
+        citiesHasNullFakeDataSource = CitiesHasNullFakeDataSource()
```

Comment:
Create an interactor for each data source:
```suggestion
        fakeDataSource = FakeDataSource()
        mixedValidDataInteractor = GetMostSuitableCityInteractor(fakeDataSource)
        citiesEmptyList = CitiesEmptyList()
        emptyDataInteractor = GetMostSuitableCityInteractor(citiesEmptyList)
        citiesHasNullFakeDataSource = CitiesHasNullFakeDataSource()
        nullDataInteractor = GetMostSuitableCityInteractor(citiesHasNullFakeDataSource)
```
Then use them in the test cases.

Reply:
Done